### PR TITLE
Rename Constructor -> Parameterized Job

### PR DESCRIPTION
**Pull request by schmichael**
```
ConstructorConfig -> ParameterizedJobConfig
Job.Constructor   -> Job.ParameterizedJob
```

And in the config:
```
constructor -> parameterized_job
```

### CODE REVIEW

**Comment by jippi** [comment]
just a personal note, `parameterized` is very long and for non-english kinda tricky to type out vs `constructor` - are any other names considered?

File: nomad/structs/diff.go [hunk 1]
```diff
@@ -641,20 +640,21 @@ func vaultDiff(old, new *Vault, contextual bool) *ObjectDiff {
...
 }
 
-// constructorDiff returns the diff of two constructor objects. If contextual
-// diff is enabled, all fields will be returned, even if no diff occurred.
-func constructorDiff(old, new *ConstructorConfig, contextual bool) *ObjectDiff {
+// parameterizedJobDiff returns the diff of two parameterized job objects. If
+// contextual diff is enabled, all fields will be returned, even if no diff
+// occurred.
+func parameterizedJobDiff(old, new *ParameterizedJobConfig, contextual bool) *ObjectDiff {
 	diff := &ObjectDiff{Type: DiffTypeNone, Name: "Constructor"}
```
Comment (dadgar) [inline nomad/structs/diff.go]: Constructor -> Parameterized

File: jobspec/test-fixtures/constructor.hcl [hunk 2]
```diff
@@ -1,5 +1,5 @@
-job "constructor" {
-    constructor {
+job "parameterized_job" {
```
Comment (dadgar) [inline jobspec/test-fixtures/constructor.hcl]: File name

**Comment by schmichael** [comment]
@jippi Sorry it's a bit of a pain. We explored a few options and felt like "parameterized" best captured what we were trying to accomplish.

File: nomad/job_endpoint_test.go [hunk 3]
```diff
@@ -797,7 +797,7 @@ func TestJobEndpoint_Evaluate_Constructor(t *testing.T) {
 	// Create the register request
 	job := mock.Job()
 	job.Type = structs.JobTypeBatch
-	job.Constructor = &structs.ConstructorConfig{}
+	job.ParameterizedJob = &structs.ParameterizedJoConfig{}
```
Comment (nkvoll) [inline nomad/job_endpoint_test.go]: This seems to be a simple typo? Should be `Job`?

Comment (dadgar) [inline nomad/job_endpoint_test.go]: Good catch!

**Pull request merged**